pkg/cloud/amazon/common.go

```diff
@@ -69,10 +69,15 @@ func ResolveRegionWithoutOptions() (string, error) {
 func ParseContext(context string) (string, string, error) {
```

Comment:
TBH,  I don't think this is a valid approach regardless. There is no guarantee on how the context name looks like. A user might change this at will and there is no guarantee that the provider changes this at will.

That said, there is  an `aws ecs describe-clusters ` which gives you cluster information. It contains the cluster arn which I think is more reliable to parse. Maybe there is even another method one can call to get the dedicated information one needs. 

So arn might be the way to go, but imo we also need to get away from relying on the context name altogether.

@dgozalo WDYT?

Reply:
The problem with `aws ecs/eks describe-clusters` is that you may get every single cluster created in a given region unless you provide a cluster name, which is exactly what we are trying to find here.

I do agree that parsing the ARN should be the way to go as it's not subject to change at all, the problem is getting the ARN of the currently connected cluster.

I think we can keep this solution until we find a more reliable way to obtain this information. We are also parsing the context in the GKE clusters and I can't find of any other solution right now.

Reply:
> The problem with aws ecs/eks describe-clusters is that you may get every single cluster created in a given region unless you provide a cluster name, which is exactly what we are trying to find here.

Maybe you could call `list-clusters` to get the list of the clusters and then iterate over them using `describe cluster` and matching on the endpoint? This way you would get to the ARN.

How else do you want to progress on this issue?

Reply:
I've finished the implementation of ARN regex testing, but I'm going to work on the list/describe clusters option to see if i can get that to work. My knowledge of Go is somewhat limited, but i'm going to give it a try anyway.

Reply:
> but I'm going to work on the list/describe clusters option to see if i can get that to work.

Nice. If we could get it to work that would be awesome.

Reply:
I've merged in changes to support this workflow, and reorganized the regex test.